molpipeline/mol2mol/filter.py

```diff
@@ -213,3 +213,48 @@ def pretransform_single(self, value: RDKitMol) -> OptionalMol:
         if value.GetNumAtoms() == 0:
             return InvalidInstance(self.uuid, "Molecule contains no atoms.", self.name)
         return value
+
+
+class InorganicsFilter(_MolToMolPipelineElement):
+    """Filters Molecules which do not contain any organic (i.e. Carbon) atoms."""
+
+    def __init__(
+        self,
+        name: str = "InorganicsFilter",
+        n_jobs: int = 1,
+        uuid: Optional[str] = None,
+    ) -> None:
+        """Initialize InorganicsFilter.
+
+        Parameters
+        ----------
+        name: str, optional (default: "InorganicsFilter")
+            Name of the pipeline element.
+        n_jobs: int, optional (default: 1)
+            Number of parallel jobs to use.
+        uuid: str, optional (default: None)
+            Unique identifier of the pipeline element.
+        """
+        super().__init__(name=name, n_jobs=n_jobs, uuid=uuid)
+
+    def pretransform_single(self, value: RDKitMol) -> OptionalMol:
+        """Invalidate molecules not containing a carbon atom.
+
+        Parameters
+        ----------
+        value: RDKitMol
+            Molecule to check.
+
+        Returns
+        -------
+        OptionalMol
+            Molecule if it contains carbon, else InvalidInstance.
+        """
+        atomic_numbers = [atom.GetAtomicNum() for atom in value.GetAtoms()]
+        if 6 not in atomic_numbers:  # Check for Carbon
+            return InvalidInstance(self.uuid, "Molecule is not organic.", self.name)
+        if len(atomic_numbers) <= 3:
+            smiles = Chem.MolToSmiles(value)
+            if smiles in ["O=C=O", "[O+]#[C-]"]:  # CO2 and CO are not organic
```

Comment:
The list of SMILES strings you test against would be better stored more globally, not only in this if statement. This makes it easier to find this list and get an overview of what molecules are excluded. I would create a class variable like

```
class InorganicsFilter(_MolToMolPipelineElement):
    IGNORE_SMILES = ["O=C=O", "[O+]#[C-]"]  # CO2 and CO are not organic
    def __ini__()
        ...
```